ang/workflow/action-links/controllers/workflow-duplicate.controller.js

```diff
@@ -0,0 +1,134 @@
+        workflow: _.clone(workflow)
+      };
+
+      model.workflow.id = null;
+      model.workflow.title = '';
+      model.workflow.name = '';
+
+      if (dialogService.dialogs.WorkflowDuplicate) {
+        return;
+      }
+
+      dialogService.open(
+        'WorkflowDuplicate',
+        '~/workflow/action-links/directives/workflow-list-duplicate-popup.html',
+        model,
+        {
+          autoOpen: false,
+          height: 'auto',
+          width: '40%',
+          title: 'Duplicate Workflow',
+          buttons: [{
+            text: ts('Save'),
+            icons: { primary: 'fa-check' },
+            click: function () {
+              if (ifSaveButtonDisabled(model.workflow_duplicate_form)) {
+                return;
+              }
+
+              createDuplicateWorkflow(model.workflow)
+                .then(function () {
+                  dialogService.close('WorkflowDuplicate');
+                });
+            }
+          }]
+        }
+      );
+    }
+
+    /**
+     * Check if Save button should be disabled
+     *
+     * @param {object} workflowDuplicateForm duplicate workflow form
+     * @returns {boolean} if Save button should be disabled
+     */
+    function ifSaveButtonDisabled (workflowDuplicateForm) {
+      return !workflowDuplicateForm.$valid || $scope.submitInProgress;
+    }
+
+    /**
+     * Create Duplicate Worklfow
+     *
+     * @param {object} workflow workflow object
+     * @returns {Promise} promise
+     */
+    function createDuplicateWorkflow (workflow) {
+      $scope.submitInProgress = true;
+
+      var promise = saveDuplicateWorkflow(workflow)
+        .then(function () {
+          $rootScope.$broadcast('workflow::list::refresh');
+        })
+        .catch(function (error) {
+          var errorMessage = error.error_code === 'already exists'
+            ? ts('This title is already in use. Please choose another')
+            : error.error_message;
+
+          return $q.reject({
+            error_message: errorMessage
+          });
+        })
+        .finally(function () {
+          $scope.submitInProgress = false;
+        });
+
+      return crmStatus({
+        start: $scope.ts('Saving Workflow...'),
+        success: $scope.ts('Saved')
+      }, promise);
+    }
+
+    /**
+     * Save duplicate workflow by calling the API
+     *
+     * @param {object} workflow workflow object
+     * @returns {Promise} Promise
+     */
+    function saveDuplicateWorkflow (workflow) {
+      workflow.name = generateWorkflowName(workflow.title);
+      workflow.sequential = true;
+
+      return civicaseCrmApi('CaseType', 'create', workflow).then(function (workflowData) {
+        return workflowData.values[0];
+      });
+    }
+
+    /**
+     * Generate workflow `name` from the sent text
+     *
+     * @param {string} text string from which name would be generated
+     * @returns {string} name of the workflow
+     */
+    function generateWorkflowName (text) {
```

Comment:
shouldn't this be generated by the API?

Reply:
The original logic was added on the FE by core, and not in api level. So following the same structure
https://github.com/civicrm/civicrm-core/blob/master/ang/crmCaseType.js#L701

Reply:
*sad noises*